components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -71,16 +74,52 @@ public boolean isSigned(RequestObject requestObject) {
     }
 
     @Override
-    public boolean validateSignature(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) throws
-            RequestObjectException {
+    public boolean validateSignature(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) {
 
+        Certificate certificate = null;
+        boolean isVerified = false;
+        String jwksUri = StringUtils.EMPTY;
         SignedJWT jwt = requestObject.getSignedJWT();
-        Certificate certificate =
-                getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
-        boolean isVerified = isSignatureVerified(jwt, certificate);
-        requestObject.setIsSignatureValid(isVerified);
-        return isVerified;
-    }
+        try {
+            certificate =
+                    getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
+        } catch (RequestObjectException e) {
+            log.warn("Public certificate not configured for Service Provider checking whether a jwks endpoint is " +
+                    "configured for the service provider");
+        }
+        if (certificate == null) {
+            try {
+                ServiceProviderProperty[] spProperties = OAuth2Util.getServiceProvider(oAuth2Parameters.getClientId()).getSpProperties();
+                for (ServiceProviderProperty spProperty : spProperties) {
+                    if (Constants.JWKS_URI.equals(spProperty.getName())) {
+                        jwksUri = spProperty.getValue();
+                        break;
+                    }
+                }
+                // Validate the signature of the assertion using the jwks end point.
+                if (!StringUtils.isBlank(jwksUri)) {
+                    if (log.isDebugEnabled()) {
+                        String message = "Found jwks end point for service provider " + jwksUri;
+                        log.debug(message);
+                    }
+                    String jwtString = jwt.getParsedString();
+                    String alg = jwt.getHeader().getAlgorithm().getName();
+                    Map<String, Object> options = new HashMap<String, Object>();
+                    isVerified = isSignatureVerified(jwtString, jwksUri, alg, options);
+                }
+                } catch (IdentityOAuth2Exception e) {
+                    e.printStackTrace();
```

Comment:
We should either log the exception or wrap and throw it

Reply:
logged the exception